addons/calendar/models/calendar_event.py

```diff
@@ -535,6 +537,13 @@ def get_interval(self, interval, tz=None):
             dummy, format_time = self._get_date_formats()
             result = tools.ustr(date.strftime(format_time + " %Z"))
 
+        elif interval == 'end_day':
+            result =  babel.dates.format_date(date=end_date, format='d MMM', locale=get_lang(self.env).code)
```

Comment:
We have the format_date function 
You can use it like this:
```
format_date(self.env, end_date))
```
It will take care of the environment, the format
```
from odoo.tools.misc import format_date
...
def format_date(env, value, lang_code=False, date_format=False):
    '''
        Formats the date in a given format.

        :param env: an environment.
        :param date, datetime or string value: the date to format.
        :param string lang_code: the lang code, if not specified it is extracted from the
            environment context.
        :param string date_format: the format or the date (LDML format), if not specified the
            default format of the lang.
        :return: date formatted in the specified format.
        :rtype: string
    '''
```